src/Resources/config/commands.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Sonata\AdminBundle\Command\CreateClassCacheCommand;
+use Sonata\AdminBundle\Command\ExplainAdminCommand;
+use Sonata\AdminBundle\Command\GenerateObjectAclCommand;
+use Sonata\AdminBundle\Command\ListAdminCommand;
+use Sonata\AdminBundle\Command\SetupAclCommand;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ReferenceConfigurator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    // Use "service" function for creating references to services when dropping support for Symfony 4.4
+    // Use "param" function for creating references to parameters when dropping support for Symfony 5.1
+    $containerConfigurator->services()
+
+        // NEXT_MAJOR: Remove this service.
+        ->set(CreateClassCacheCommand::class, CreateClassCacheCommand::class)
+            ->public()
+            ->tag('console.command', [
+                'command' => 'cache:create-cache-class',
+            ])
+            ->deprecate('The "%service_id%" service is deprecated since version sonata-project/admin-bundle 3.39.0 and will be removed in 4.0.')
```

Comment:
```suggestion
            ->deprecate('The "%service_id%" service is deprecated since sonata-project/admin-bundle 3.39.0 and will be removed in 4.0.')
```